File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/customer/CustomerStatus.java
package br.com.deltaglobalbank.customers.domain.customer;

import java.util.Arrays;
import java.util.Locale;

public enum CustomerStatus {
    ACTIVE,
    INACTIVE;

    public String toDatabaseValue() {
        return name().toLowerCase(Locale.ROOT);
    }

    public static CustomerStatus fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(status -> status.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("invalid costumer status: " + value));
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/customer/CustomerStatus.kt
package br.com.deltaglobalbank.customers.domain.customer

enum class CustomerStatus {
    ACTIVE,
    INACTIVE;

    fun toDatabaseValue() : String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): CustomerStatus =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("invalid costumer status: $value")
    }
}